### DISCUSSION TITLE
Reticulum and interface to Qortal

### DISCUSSION THREAD

**Discussion by karl-baker** [markqvist/Reticulum#178] [General]
Are there any plans on expanding the Reticulum interfaces to include a configuration for access/communication to and from the Qortal ecosystem?

The Qortal project is looking to expand and include mesh networking, see https://wiki.qortal.org/doku.php?id=mesh_networking for details.

Are there others using Reticulum that would be interested in the same interface?

**Comment by markqvist** [comment]
Hi @karl-baker! Interesting topic.

As far as I can see (and correct me if I am wrong), the so-called "Qortal Mesh Networking Technology" that page is talking about is currently just an idea, and does not actually *exist*, which makes it a bit difficult to create an interface for, and even more so to actually use as functional data transport ;)

It seems the other way around would be more suitable at this point. Since Reticulum already works, and can provide both local and wide-area data transport over practically any kind of wired or wireless connection, Qortal could definitely use Reticulum to act as the base-layer mesh fabric for their applications and larger/longer term network ambitions.

**Reply by karl-baker** [reply]
That is correct, it is currently something that is nonexistent. The Qortal team is interested in utilizing Reticulum in order to build their mesh network.

**Reply by markqvist** [reply]
Aha, I think I misunderstood it initially then.

I understood it as a request for a adding an interface to Reticulum, allowing it to utilise Qortals network as a carrier.

**Reply by crowetic** [reply]
Yes, hello @markqvist - very happy to have this topic finally opened.

This is crowetic with Qortal nice to meet you. :) I have been looking to get in touch with you for a while now. 

Yes, you are 100% correct, at this point the Qortal Mesh is simply a concept. When someone pointed out reticulum to me, I was like 'OMG, someone actually did it already, it's ideal!' 

We were thinking that we would have to develop it ourselves, which obviously is no simple task. So when I saw reticulum I was mind-blown to see that exactly what I wanted built correctly in every way, was already done! 

So what my goal is now, is to make Qortal communications work over reticulum. This would also provide encryption to Qortal's sync communications, and allow us to connect without needing IP addresses once the mesh gets a bit bigger, until then it will utilize 'internet gateways' to connect to the rest of the network.

I was wondering what a setup like this would look like to allow Qortal to connect over reticulum, but I haven't had much of a chance to start messing around with it yet.

I HAVE run a reticulum node as a test, and connected to the testnet, but then formatted the computer with the node on it and haven't reconfigured it yet.

Anyway, I'm glad we got this discussion started, let me know if you're able to assist at all, even if just in theory, and if there's a way I can have personal communications with you. Thank you so much for your work and response!

**Comment by markqvist** [comment]
Hm. Just read the primer material and some other parts of the wiki. To put it diplomatically, I remain unimpressed.

From their primary project description on github:

> Qortal Project Core - decentralize the world  [...] complete infrastructure for the future blockchain based Internet

Why anyone would want a *blockchain*-based *Internet* is completely beyond me. Seems like the straight way to a completely *centralised*, highly failure-prone and very fragile piece of infrastructure.

It seems to run directly contrary to their "decentralize the world" tagline. I guess what they really mean is *re*-centralise the world, onto our platform.

**Reply by karl-baker** [reply]
If you would like, I can put you in contact with the founder of the Qortal project. You can then discuss the ideological differences and reach a consensus or agree to disagree.

Let me know, I'd be more than happy to see the two of you begin a dialogue.

**Reply by markqvist** [reply]
I only spent a short time superficially reading the basic stuff from their wiki, so I will happily admit that I might be completely missing something smart and brilliant here. There was just too high a proportion of what I did read, that seemed self contradictory or poorly thought through. It seems like there's some really good ideals and intentions in there, though.

And just because I might not "get" a project, or even see it as backwards, I completely encourage anyone to use Reticulum if it can be useful in their projects.

I might be happy to state my *opinion* on, and question something, but ultimately, I won't and can't be the judge of what is useful or not, so if they want to use it, they should just go ahead and do so!

I sincerely hope all kinds of people, with all kinds of ideas, will build all kinds of projects I didn't couldn't even dream up myself. In that regard, I am trying to make Reticulum as useful a tool as possible.

**Comment by markqvist** [comment]
For the record, my previous statement:

> Why anyone would want a blockchain-based Internet is completely beyond me. Seems like the straight way to a completely centralised, highly failure-prone and very fragile piece of infrastructure.

was informed by the assumption that the blockchain dependency would run straight to the base-layer communications protocols, which I think is a very bad idea. When someone says the word "internet", I will tend, by the bias of where my work lies, to understand that as a talking about base-layer protocols, such as Reticulum or IPv4/v6.

I understand now that Qortal is currently focusing on everything *above* that. From my, admittedly still limited, understanding of Qortals internals, it does look like a really interesting approach to the problems they are tackling.

**Reply by crowetic** [reply]
Yes, I would be more than happy to explain our current methods to you, and give you a better understanding. We are NOT utilizing blockchain tech in ANYTHING like any other project's methods.

Our technology is completely different and built from scratch for a reason.

I believe you'll like it once you gain a bit better understanding of the concept. I also believe that reticulum and Qortal have a great future together. :)

Trust me when I say, I can't STAND most 'crypto projects' so much so that I even hate the correlation between Qortal and the word 'crypto' at all... The majority of the projects out there are complete jokes, cash grabs, and completely worthless. Doing everything at the best, semi-decentralized.

Qortal is COMPLETELY different in every way, and done 100% decentralized with a completely different type of blockchain, with a TON more functionality.

Our data network, QDN, is also completely different, and does NOT store anything other than a single hash (per registered name) for the files on the chain, so that they're SECURED by the chain, but not STORED on it. Making it actually SCALABLE to the level it needs to be. Most other projects half-ass nearly everything, making semi-decentralized pieces of crap IMO.

We have been working on this project for nearly 10 years, ensuring that every aspect was done completely correctly. Would love to chat with you a bit in private if possible. I attempted to do so over your reticulum testnet, but didn't know how to actually reach you there. (I admit I didn't put a TON of time into figuring it out, but yea...)

I really think reticulum+Qortal has a VERY bright future, and I believe you'll like the project once you fully understand it and see why we did everything the way we did. 

The main component we're missing, is the protocol layer Mesh Network connectivity that reticulum provides! I believe Qortal could be the ideal 'frontend' for reticulum, and provide a plethora of unique and awesome additional functionality when combined with one another.

**Reply by crowetic** [reply]
and yes, you're completely correct, we are currently focusing on replacing the infrastructure ABOVE the protocol level, then we aim to get down to the protocol level. Which is where reticulum I think fits perfectly. It is VERY difficult to find others that wholeheartedly believe in TRUE open source projects. I couldn't believe it when I saw reticulum, and it was everything I could ever want, it was literally the EXACT concept I had envisioned for the protocol-level Mesh Network connectivity I wanted for Qortal, it was a dream come true. 

Thank you SO MUCH for doing what you do, and for the way you do it. I look forward very much to being able to make massive contributions to your project in the future when I am able. In any possible way.

**Comment by karl-baker** [comment]
I have been in contact with the founder of the Qortal project. He is very interested in beginning a dialog with you.  His name is Jason Crowe, and you can reach him on Telegram at @crowetic.

This may turn out to be very prosperous for both sides.

**Reply by crowetic** [reply]
Thank you very much for initiating this communication Karl, I don't know why I didn't think of github as a good method to get in touch, sometimes the most basic things don't pop out at me directly. heh.

**Reply by karl-baker** [reply]
You're more than welcome Jason! Happy to see that a dialog is beginning.
